src/eva/vision/data/datasets/segmentation/consep.py

```diff
@@ -0,0 +1,150 @@
+    def class_to_idx(self) -> Dict[str, int]:
+        return {label: index for index, label in enumerate(self.classes)}
+
+    @override
+    def filename(self, index: int) -> str:
+        sample_idx, slice_index = self._indices[index]
+        sample_dir = self._samples_names[sample_idx]
+        return os.path.join(sample_dir, f"{slice_index}-ct.nii.gz")
+
+    @override
+    def prepare_data(self) -> None:
+        _validators.check_dataset_exists(self._root, True)
+
+    @override
+    def configure(self) -> None:
+        self._samples_names = self._fetch_samples_names()
+
+    @override
+    def validate(self) -> None:
+        _validators.check_dataset_integrity(
+            self,
+            length=self._expected_dataset_lengths.get(self._split),
+            n_classes=len(self.classes),
+            first_and_last_labels=((self.classes[0], self.classes[-1])),
+        )
+
+    @override
+    def __len__(self) -> int:
+        return len(self._samples_names)
+
+    @override
+    def load_image(self, index: int) -> tv_tensors.Image:
+        image_path = self._get_image_path(index)
+        image = io.read_image_as_tensor(image_path)
+        return tv_tensors.Image(image, dtype=torch.float32)
+
+    @override
+    def load_mask(self, index: int) -> tv_tensors.Mask:
+        ground_truth = io.read_mat(self._get_mask_path(index))
+        mask = ground_truth["type_map"]
+        return tv_tensors.Mask(mask, dtype=torch.int64)
+
+    def get_file_path(self, index: int, image_or_label: str, file_extension: str) -> str:
+        """Returns the corresponding image path."""
+        sample_name = self._samples_names[index]
+        if self._split == "train":
+            split = "Train"
+        elif self._split == "val":
+            split = "Test"
+        else:
+            if index <= self._expected_dataset_lengths["train"]:
+                split = "Train"
+            else:
+                split = "Test"
+        return os.path.join(self._root, split, image_or_label, sample_name + file_extension)
+
+    def _get_image_path(self, sample_index: int) -> str:
+        """Returns the corresponding image path."""
+        return self.get_file_path(sample_index, "Images", ".png")
+
+    def _get_mask_path(self, sample_index: int) -> str:
+        """Returns the directory of the corresponding masks."""
+        return self.get_file_path(sample_index, "Labels", ".mat")
+
+    def _get_number_of_slices_per_sample(self, sample_index: int) -> int:
+        """Returns the total amount of slices of a sample."""
+        image_path = self._get_image_path(sample_index)
+        image_shape = io.fetch_nifti_shape(image_path)
+        return image_shape[-1]
+
+    def _fetch_samples_names(self) -> List[str]:
+        """Returns the name of all the samples of the selected dataset split."""
+
+        def _samples_dirs_split(split: str) -> List[str]:
+            split_dir = os.path.join(self._root, f"{split}/Images")
+            return [
```

Comment:
Nit: All images seem to be `.png`, so using glob might be safer / more concise:
something like 
```python
[os.path.basename(f) for f in glob.glob(os.path.join(self._root, f"{split_dir}/*.png"))]
```